server/src/main/java/com/objectcomputing/checkins/services/file/FileServicesImpl.java

```diff
@@ -0,0 +1,226 @@
+        }
+    }
+
+    @Override
+    public HttpResponse<FileInfoDTO> uploadFile(@NotNull UUID checkInID, @NotNull CompletedFileUpload file) {
+
+        CheckIn checkIn = checkInServices.read(checkInID);
+        validate(checkIn == null, "Unable to find checkin record with id %s", checkInID);
+        validate((file.getFilename() == null || file.getFilename().equals("")), "Please select a file before uploading.");
+
+        // create folder name in the format name-date
+        String subjectName = memberProfileServices.getById(checkIn.getTeamMemberId()).getName();
+        String sb = subjectName.concat(LocalDate.now().toString());
+        sb = sb.replaceAll("\\s", "");
+        final String directoryName = sb;
+
+        try {
+            Drive drive = googleDriveAccessor.accessGoogleDrive();
+            validate(drive == null, "Unable to access Google Drive");
+
+            /* check if folder already exists on google drive
+            * if exists, return folderId and name
+            * else, create a new folder in the format name-date */
+            FileList driveIndex = drive.files().list().setFields("files(id, name)").execute();
+            File folderOnDrive = driveIndex.getFiles().stream()
+                                    .filter(s -> directoryName.contains(s.getName()))
+                                    .findFirst()
+                                    .orElse(createNewDirectoryOnDrive(drive, directoryName));
+
+            // set file metadata
+            File fileMetadata = new File();
+            fileMetadata.setName(file.getFilename());
+            fileMetadata.setMimeType(file.getContentType().orElse(MediaType.APPLICATION_OCTET_STREAM_TYPE).toString());
+            fileMetadata.setParents(Collections.singletonList(folderOnDrive.getId()));
+
+            //upload file to google drive
+            InputStreamContent content = new InputStreamContent(file.getContentType().toString(), file.getInputStream());
+            File uploadedFile = drive.files().create(fileMetadata, content)
+                                .setSupportsAllDrives(true)
+                                .setFields("id, size, name")
+                                .execute();
+
+            //create record in checkin-document service
+            CheckinDocument cd = new CheckinDocument(checkInID, uploadedFile.getId());
+            checkinDocumentServices.save(cd);
+
+            emailSender.sendEmail("New Check-in Notes", "New check-in notes have been uploaded. Please check the Google Drive folder.");
+
+            return HttpResponse
+                    .status(HttpStatus.OK)
+                    .body(setFileInfo(uploadedFile, cd));
+        } catch (IOException e) {
+            LOG.error("Unexpected error processing file upload.", e);
+            return HttpResponse.status(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @Override
+    public HttpResponse<?> deleteFile(@NotNull String uploadDocId) {
+
```

Comment:
Should also only be able to be done by the folk above...again...can be part of a new story.

Reply:
updated in the latest commit.